### PR TITLE
Make attributes case-sensitive; change case

### LINKED ISSUE #2: Keys should be all lower case (or at least the case should be required)
The spec currently allows keys to be any case, but recommends mixed case. This is a bad idea. For one, it's much harder to write a schema for, and therefore harder to both validate and autocomplete. It also means that when a consumer reads the CPS, if their json implementation doesn't support case-insensitive lookup (I have yet to see one that does), they have to recursively walk the keys, lower case them, and hope there's no collisions.

### PR DESCRIPTION
**Pull request by mwoehlke**
In line with community feedback (see #2), make JSON keys case sensitive, and use all lower-case with underscores rather than title-case with hyphens. This mostly affects attribute names, but also some attributes that use language names as keys for language-specific values.

Needless to say, this is a major compatibility break.

Closes #2.

### CODE REVIEW

File: schema.rst [hunk 1]
```diff
@@ -292,14 +290,14 @@ shall be explicitly undefined (e.g. ``-UFOO`` passed to the compiler).
 A map may be used instead to give different values
 depending on the language of the consuming source file.
 In this case, the build tool shall select the list
-from the map whose key matches the (case-insensitive) language
+from the map whose key matches the (case-sensitive) language
```
Comment (dcbaker) [inline schema.rst:293]: Would it be better to include lower case here, `(lower-case, case-sensitive)`?

File: schema.rst [hunk 2]
```diff
@@ -429,12 +427,12 @@ use of this attribute is discouraged.
 Specifies the ABI language or languages of a static library
 (`Type`_ :string:`"archive"`).
 Officially supported (case-insensitive) values are
-:string:`"C"` (no special handling required) and
-:string:`"C++"` (consuming the static library
+:string:`"c"` (no special handling required) and
+:string:`"cpp"` (consuming the static library
```
Comment (dcbaker) [inline schema.rst:431]: This still says case-insensitive

**Review by dcbaker — COMMENTED** [review]
I still see some cases where string values may be passed, but with recommended values. It seems like we should remove the case-insensitivity of those too?

File: schema.rst (hunk 1 above)
Comment (mwoehlke) [reply schema.rst]: Done.

File: schema.rst (hunk 2 above)
Comment (mwoehlke) [reply schema.rst:431]: Yes, I am still torn on that. For *key* names, there's a technical reason to enforce case-sensitivity. For *values*, doing likewise seems like gratuitous violation of the [Robustness Principle](https://en.wikipedia.org/wiki/Robustness_principle)... which is why I originally specified case-insensitivity in the first place and, if not for said technical issues, would prefer to have retained it.

I'm not saying I'm dead-set on this point, just... reluctant.

**Review by bretbrownjr — CHANGES_REQUESTED** [review]
Nice! This is definitely what we discussed changing.

Some probably overlooked details I noticed grepping around a clone of this branch:

* [ ] sample.cps
* [ ] configurations example in recommendations.rst

And possibly:

* [ ] recommendations.rst:138
* [ ] schema.rst:445,476,556
* [ ] searching.rst:88

If you'd rather turn these into a tracking issue, merge this PR, and circle back with another PR, that works for me too.

File: schema.rst (hunk 2 above)
Comment (mwoehlke) [reply schema.rst:431]: > Also, I think you could add the Closes: #2 or link the issue.

It was already linked, but good point on adding an auto-close tag; done!

**Comment by mwoehlke** [comment]
> Some probably overlooked details I noticed

Yup, I was noticing those also; thanks! (I found some others that didn't make your list, that I'll be fixing as well...)

**Comment by mwoehlke** [comment]
Oops, forgot `sample.cps`. (Hopefully I found everything else, though!)

**Review by bretbrownjr — APPROVED** [review]
I like this change. It's possible there's another detail or two we missed, but it should be easy enough to do a quick patch release to resolve those.

**Comment by mwoehlke** [comment]
Thanks, @bretbrownjr! Since the EE meeting is tomorrow, I'll give this until then in case there is other feedback and plan on merging it after that.

**Comment by bretbrownjr** [comment]
We didn't explicitly discuss this in the C++ Ecosystem Evolution meeting, but it has been two days.

I'm happy to help merge PRs like this, but I don't have permissions at the moment.

**Pull request merged**